src/main/java/uk/gov/justice/probation/courtcasematcher/messaging/HearingProcessor.java

```diff
@@ -123,13 +123,12 @@ private void updateAndSave(final Hearing hearing) {
     }
 
     Hearing assignUuids(Hearing hearing) {
-        // Apply the new case ID
-        final var caseId = UUID.randomUUID().toString();
-        var updatedHearing = hearing.withCaseId(caseId).withHearingId(caseId);
+        var updatedHearing = hearing.withHearingId(UUID.randomUUID().toString());
 
         // We want to retain the LIBRA case no if present
         if (hearing.getCaseNo() == null) {
-            updatedHearing = updatedHearing.withCaseNo(caseId);
+            updatedHearing = updatedHearing.withCaseNo(hearing.getCaseId());
```

Comment:
Do we have a test for this behaviour? I know that it hasn't changed here

Reply:
Added some unit tests